src/Tracking/Events/Activated.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tracking\Events;
+
+/**
+ * This class adds actions to track when the extension is activated.
+ * *
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tracking
+ */
+class Activated extends BaseEvent {
+
+	/**
+	 * The page where activation with a source can occur.
+	 */
+	public const ACTIVATION_PAGE = 'plugin-install.php';
+
+	/**
+	 * The query parameters used to determine activation source details.
+	 */
+	public const SOURCE_PARAMS = [
+		'traffic_source',
+		'traffic_type',
+	];
+
+	/**
+	 * Register the tracking class.
+	 */
+	public function register(): void {
+		add_action( 'woocommerce_gla_extension_activated', [ $this, 'maybe_track_activation_source' ] );
+	}
+
+	/**
+	 * Track when the extension is activated from a source.
+	 *
+	 * @param array $server_vars The server variables to check for source information.
+	 */
+	public function maybe_track_activation_source( array $server_vars = [] ): void {
+		// Default to $_SERVER
+		if ( empty( $server_vars ) ) {
+			$server_vars = $_SERVER;
```

Comment:
Do we need to check for the server variables here? I'm not sure if it's suitable to have it as an argument for this method either. It seems we can inject the server vars into the constructor of this class since it's not something that will change during the request.

Reply:
a7b4bc44bcb4d5dd507b19058f6f0c1373eb9fc7